server/src/main/java/io/crate/statistics/TransportAnalyzeAction.java

```diff
@@ -84,26 +86,45 @@ public final class TransportAnalyzeAction {
     private final TransportService transportService;
     private final Schemas schemas;
     private final ClusterService clusterService;
+    private final AtomicReference<CompletableFuture<Samples>> runningAnalyze = new AtomicReference<>();
+    private final Executor executor;
 
     @Inject
     public TransportAnalyzeAction(TransportService transportService,
                                   ReservoirSampler reservoirSampler,
                                   Schemas schemas,
                                   ClusterService clusterService,
-                                  TableStats tableStats) {
+                                  TableStats tableStats,
+                                  ThreadPool threadPool) {
         this.transportService = transportService;
         this.schemas = schemas;
         this.clusterService = clusterService;
+        this.executor = threadPool.executor(ThreadPool.Names.SEARCH);
+
         transportService.registerRequestHandler(
             FETCH_SAMPLES,
-            ThreadPool.Names.SEARCH,
+            ThreadPool.Names.SAME,
```

Comment:
Why do you change this? This would result running the sample collection inside the netty thread which we should avoid for long running tasks.

Reply:
Ah sorry didn't see, it will dispatch later, ignore my comment ;)